File: Task/Sleeping-Beauty-problem/C++/sleeping-beauty-problem.cpp
#include <iostream>
#include <random>

int main() {
    std::cout.imbue(std::locale(""));
    const int experiments = 1000000;
    std::random_device dev;
    std::default_random_engine engine(dev());
    std::uniform_int_distribution<int> distribution(0, 1);
    int heads = 0, wakenings = 0;
    for (int i = 0; i < experiments; ++i) {
        ++wakenings;
        switch (distribution(engine)) {
        case 0: // heads
            ++heads;
            break;
        case 1: // tails
            ++wakenings;
            break;
        }
    }
    std::cout << "Wakenings over " << experiments
              << " experiments: " << wakenings << '\n';
    std::cout << "Sleeping Beauty should estimate a credence of: "
              << double(heads) / wakenings << '\n';
}

File: Task/Sleeping-Beauty-problem/Swift/sleeping-beauty-problem.swift
let experiments = 1000000
var heads = 0
var wakenings = 0
for _ in (1...experiments) {
    wakenings += 1
    switch (Int.random(in: 0...1)) {
    case 0:
        heads += 1
    default:
        wakenings += 1
    }
}
print("Wakenings over \(experiments) experiments: \(wakenings)")
print("Sleeping Beauty should estimate a credence of: \(Double(heads) / Double(wakenings))")
